src/header/js/header-menus.js

```diff
@@ -237,6 +244,68 @@ export default ({variant = 'desktop'}) => {
 		);
 	});
 
+	function resetFlyoutDimensions(parentNode) {
+		const subMenus = document.querySelectorAll('.esri-header-menus-submenu');
+		const parent = (parentNode !== 'disabled' && parentNode !== 'init') && parentNode.getAttribute('data-parent');
+		const parentElement = document.querySelector(`#${parent}`);
+
+		if (parentNode === 'init') {
+			const listItems = [].slice.call(document.querySelectorAll('.esri-header-menus-flyout'));
+
+			if (listItems.length) {
+				listItems.forEach((fly) => {
+					const catItem = [].slice.call(fly.querySelectorAll('.esri-header-menus-flyout--categories-item'));
+					const catItemParent = document.querySelector(`#${catItem[0].getAttribute('data-parent')}`);
+	
+					const listItems = [].slice.call(fly.querySelectorAll('.esri-header-menus-flyout--list-items'));
+					const listColType = listItems[0].getAttribute('data-coltype');
+	
+					if (listColType === '1') {
+						catItemParent.setAttribute('data-single', '');
+					}
+				});
+			}
+		} else if (parentNode === 'disabled') {
+			subMenus.forEach((menu) => {
+				menu.removeAttribute('data-single');
+			});
+		} else {
+			parentElement.setAttribute('data-single', '');
+		}
+	}
+
+	function resetFlyoutState() {
+		const flyoutCategories = [].slice.call(document.querySelectorAll('.esri-header-menus-flyout--categories-item'));
+		const flyoutList = [].slice.call(document.querySelectorAll('.esri-header-menus-flyout--list-items'));
+
+		if (flyoutList.length) {
+			flyoutList.forEach((list, index) => {
+				flyoutCategories[index].setAttribute('aria-current', 'false');
+				list.setAttribute('aria-current', 'false');
+				if ((list.hasAttribute('data-id') && list.getAttribute('data-id') === '0') && 
+						(flyoutCategories[index].hasAttribute('data-id') && flyoutCategories[index].getAttribute('data-id') === '0')
```

Comment:
you don't need the hasAttribute checks here - just the getAttribute checks will do the same thing